west: blobs: fetch exit immediately on HTTP error

noahp opened the pull request:
If an HTTP error occurs during `west blobs fetch`, the response was written to a file (even if the response body is empty), and then the checksum validation fails, which can be somewhat confusing.

Add an immediate error message and exit-with-error-code when the HTTP request fails.

Tested by modify a blob manifest to have an invalid URL:

```bash
# test with invalid URL
❯ west blobs fetch nrf_wifi
Fetching blob nrf_wifi: /home/noah/dev/zephyrs/main/modules/lib/nrf_wifi/zephyr/blobs/wifi_fw_bins/default/nrf70.bin
ERROR: HTTP error occurred: 404 Client Error: Not Found for url: https://github.com/nrfconnect/sdk-nrfxlib/raw/863f1b54a563d2139cc26514a56817feb0f91e61/zzzz/nrf_wifi/bin/zephyr/default/nrf70.bin

# test with networking disabled
❯ west blobs fetch nrf_wifi
Fetching blob nrf_wifi: /home/noah/dev/zephyrs/main/modules/lib/nrf_wifi/zephyr/blobs/wifi_fw_bins/default/nrf70.bin
ERROR: An error occurred: HTTPSConnectionPool(host='github.com', port=443): Max retries exceeded with url: /nrfconnect/sdk-nrfxlib/raw/863f1b54a563d2139cc26514a56817feb0f91e61/zzzz/nrf_wifi/bin/zephyr/default/nrf70.bin (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x7bef7c4b9a90>: Failed to resolve 'github.com' ([Errno -3] Temporary failure in name resolution)"))
```

pdgendt commented on scripts/west_commands/fetchers/http.py:
```diff
@@ -5,6 +5,8 @@
 import requests
 
 from west import log
+import sys
+import os
```
Keep Python imports together (sorted)

```suggestion
import os
import requests
import sys

from west import log
```

pdgendt commented on scripts/west_commands/fetchers/http.py:
```diff
@@ -16,5 +18,14 @@ def schemes(cls):
+        except requests.exceptions.RequestException as e:
+            log.err(f'An error occurred: {e}')
+            sys.exit(os.EX_DATAERR)
+        else:
+            open(path, "wb").write(resp.content)
```
No need to put it in an `else`, also prefer using a context

```suggestion
        with open(path, "wb") as f:
            f.write(resp.content)
```

pdgendt commented:
You need to squash your commits into a single one, and force push.

noahp commented:
> You need to squash your commits into a single one, and force push.

All set, thank you!

noahp committed:
west: blobs: fetch exit immediately on HTTP error

If an HTTP error occurs during `west blobs fetch`, the response was written
to a file (even if the response body is empty), and then the checksum
validation fails, which can be somewhat confusing.

Add an immediate error message and exit-with-error-code when the HTTP
request fails.

Tested by modify a blob manifest to have an invalid URL:

```bash
# test with invalid URL
❯ west blobs fetch nrf_wifi
Fetching blob nrf_wifi: .../wifi_fw_bins/default/nrf70.bin
ERROR: HTTP error occurred: 404 Client Error: Not Found for url: ...

# test with networking disabled
❯ west blobs fetch nrf_wifi
Fetching blob nrf_wifi: .../wifi_fw_bins/default/nrf70.bin
ERROR: An error occurred: HTTPSConnectionPool(host='git.... \
  Max retries exceeded with url: \
  .../zzzz/nrf_wifi/bin/zephyr/default/nrf70.bin
...
```

Signed-off-by: Noah Pendleton <noah.pendleton@gmail.com>
